### ISSUE TITLE
Books added multiple times

### ISSUE THREAD

**Issue by mhutti1** [kiwix/kiwix-android#778]
### Are you ready to report?
   - Your issue may already be reported! Please search on the [issue track](../) before creating one.
   - Ensure you have tested with the last (dev) version of the software.  (If not, and if possible, run your test again with the most recent version available.)

### Environment
  * version of the software you use : 3.0
  * device / operating system : 8.0
  
### The Bug:

Originally I crashed with a conncurent array modifcation exception at https://github.com/kiwix/kiwix-android/blob/f47bd50f373911c0cba7a5b051cda9a4090ebab5/app/src/main/java/org/kiwix/kiwixmobile/data/local/dao/BookDao.java#L129

Fixing this bug with this code:
```
  public void scan(String defaultPath) {
    // Start custom file search

    Thread fs = new Thread(() -> {
      scanFileSystem(defaultPath);
    });
    fs.start();
    // Star mediastore search
    new Thread(() -> {
      try {
        scanMediaStore();
        fs.join();
        checkCompleted();
      } catch (InterruptedException e) {
        checkCompleted();
      }
    }).start();
  }
```
Here https://github.com/kiwix/kiwix-android/blob/974ec85da9f32b09a74bea23d74474153c3bdcd4/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileSearch.java

results in duplicate items. I think this is a flaw in the new file search code as it didn't occur on master.

#### Steps to reproduce:

Not sure can look into if necessary.

**Comment by mhutti1** [comment]
Might be fixed by #663

**Comment by abdulwd** [comment]
@mhutti1 I think we should get rid of **ZimFileSelectFragment** as it is just duplicating the new home screen. The only additional functionality which is provided by the **ZimFileSelectFragment** is the delete feature which can easily be integrated into the new home screen.

**ZimFileSelectFragment** also creates the object of **FileSearch**. So, there will be two searches going on unnecessarily which might lead to the addition of duplicate elements.
Also, after removing the **ZimFileSelectFragment** the code in the **FileSearch** class can be moved to the **MainPresenter**.

May I remove the fragment?

**Comment by mhutti1** [comment]
This might be what we do in the long term but it is not the solution for
now. It is not difficult to avoid duplicates. At all times an insertion
should check if the zim id is already present (we should likely lock the
list to avoid concurrency issues)

**Comment by abdulwd** [comment]
Closed via #805.
